bin/internal/shared.sh

```diff
@@ -137,9 +140,14 @@ function upgrade_flutter () (
 
     # Fetch Dart...
     rm -f "$FLUTTER_ROOT/version"
+    rm -f "$FLUTTER_ROOT/.version.json"
     touch "$FLUTTER_ROOT/bin/cache/.dartignore"
     "$FLUTTER_ROOT/bin/internal/update_dart_sdk.sh"
 
+    if [ "$(basename $0)" = 'dart' ]; then
+      exit 0
+    fi
```

Comment:
I'm unfamiliar with this script. Why exit here if being invoked by dart? Might this break `/bin/dart` since `execute` never gets defined?

https://github.com/flutter/flutter/blob/db8d1a441739ed869f25919ce5e43ffacd6e0228/bin/dart#L55-L58

Reply:
Oops, meant to pull this into a different change. I'll remove this diff. As a sneak preview and to answer your question, `/bin/dart` won't ever invoke this bash script (you can only get here from //flutter/bin/dart), and we're exiting early because we don't need to try to build the flutter tool if the user is invoking dart. This fix was so (for example) you can run the analyzer if you're refactoring the tool and the tool actually doesn't compile. It helps to have a working analyzer if your code doesn't compile :)

Reply:
Done